resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -215,8 +232,10 @@ private[spark] class Client(
     sparkConf.setIfMissing("spark.driver.port", DEFAULT_DRIVER_PORT.toString)
     sparkConf.setIfMissing("spark.blockmanager.port",
       DEFAULT_BLOCKMANAGER_PORT.toString)
+    sparkConf.set("spark.ui.basePath", s"/$kubernetesAppId/$UI_PATH_COMPONENT")
```

Comment:
I guess this should only be set when `exposeIngress` is true, otherwise it would affect the current non-ingress deploy.

I just ran the integration tests and it failed at the first case which don't have `exposeIngress` set, and the test log shows the spark api returns 404.